### ISSUE TITLE
VPN fails to connect with big sized DNS filterlist

### ISSUE THREAD

**Issue by Morku** [AdguardTeam/AdguardForiOS#2157]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

- **AdGuard version:**
  -  4.4.6.998
- **Device model and storage size:**
  -  iPhone Xs Max 64gb
- **Operating system and version:**
  -  iOS 16.4.1

1. Add `https://raw.githubusercontent.com/hagezi/dns-blocklists/main/adblock/pro.plus.txt` under DNS protection
2. Enable DNS protection
3. VPN tries to connect, but fails and is never done -> no connection to internet possible
4. when disable HaGeZi's DNS list, VPN is just connecting fine again
5. enable HaGeZi's DNS list and the issue directly recovers

That the issue is related due size of the DNS list, is just speculation of my side. The issue could be something else, but over 500.000 entries is big. It also could be something in the filterlist specific.

A logfile will be send after.

### Expected Behavior

In best case a connected and working fine VPN connection with filtering.

If size is the limit/problem -> a better indictation why VPN protection won't work with the filterlist.
At current state, the user is adding the big DNS filterlist, but has no idea why it is not working.

### Actual Behavior

DNS VPN does not connect. You can see under VPN setting, how it tries to connect and even "Disconnect" to retry to connect in a loop. Check the Video.

### Video

<details><summary>Video:</summary>

https://user-images.githubusercontent.com/6276915/231964207-06111592-6f0f-4cf4-bd83-fd2e49268577.mp4

</details>

**Comment by jslawler** [comment]
Reproduced on iOS 16.4.1, iPhone 13 Pro 128gb.

**Comment by Versty** [comment]
@Morku Yes, your guess is correct, this issue is caused by the size of DNS filter list. 
iOS have a strict limit of 15MB RAM usage in the background for the VPN tunnel. 
We are currently refactoring the application code, some kind of warning for this case should definitely be added in the future versions.
